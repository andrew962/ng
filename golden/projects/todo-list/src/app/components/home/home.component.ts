import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { MatDialog, MatDialogRef } from '@angular/material/dialog';
import { Title } from '@angular/platform-browser';
import { NewItemList } from '../../models/newitem.model';
import { DialogCreateNewComponent } from '../templates/dialog-create-new.component';
import * as _ from 'underscore';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styles: [
  ]
})
export class HomeComponent implements OnInit {
  listItems: NewItemList[] = [];
  itemSelected: NewItemList;
  itemSelectToEdit: FormGroup;
  constructor(private title: Title,
    private dialog: MatDialog,
    private fb: FormBuilder) {
    this.itemSelectToEdit = fb.group({
      Title: [, Validators.required],
      Content: [],
      Uid: []
    });
  }

  ngOnInit(): void {
    this.title.setTitle('Inicio');
  }

  btnNewItemList() {
    let dialogRef = this.dialog.open(DialogCreateNewComponent);
    dialogRef.afterClosed().subscribe((res: NewItemList) => {
      if (res.Title != null) {
        this.listItems.push(res);
      }
    });
  }
  itemSelectedChange(event: NewItemList) {
    this.itemSelected = event;
    this.itemSelectToEdit.setValue({
      Title: event.Title,
      Content: event.Content,
      Uid: event.Uid
    });
  }
  btnSaveEdited() {
    let selected: NewItemList = _.findWhere(this.listItems, { Uid: this.itemSelectToEdit.controls.Uid.value });
    console.log(this.listItems, selected);
    selected.Title = this.itemSelectToEdit.controls.Title.value;
    selected.Content = this.itemSelectToEdit.controls.Content.value;
    this.itemSelected = null;
  }

}
