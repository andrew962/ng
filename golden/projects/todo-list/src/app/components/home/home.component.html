<div class="row">
    <div class="col-12 text-right">
        <button (click)="btnNewItemList()" class="btn btn-primary btn-sm"><i class="fas fa-sticky-note"></i>
      Nuevo</button>
    </div>
</div>
<div class="row">
    <div class="col-6">
        <app-list-manage [(listItems)]="listItems" (itemSelectedChange)="itemSelectedChange($event)"></app-list-manage>
    </div>
    <div class="col-6">
        <ng-container *ngIf="itemSelected != null; else noSelected">
            <form [formGroup]="itemSelectToEdit">
                <mat-form-field class="w-100">
                    <mat-label>Titulo</mat-label>
                    <input formControlName="Title" matInput placeholder="label" value="value">
                </mat-form-field>
                <mat-form-field class="w-100">
                    <mat-label>Contenido</mat-label>
                    <textarea matInput formControlName="Content"></textarea>
                </mat-form-field>
                <div class="col-12 p-0 text-right">
                    <button class="btn btn-success btn-sm" (click)="btnSaveEdited()"><i class="fas fa-save"></i></button>
                </div>
            </form>
        </ng-container>
        <ng-template #noSelected>
            <h2>No a seleccionado nada</h2>
        </ng-template>
    </div>
</div>